stablehlo/reference/Ops.cpp

```diff
@@ -224,12 +236,172 @@ Tensor evalTransposeOp(const Tensor &operand, ArrayRef<int64_t> permutation,
   return result;
 }
 
+SmallVector<Tensor> evalWhileOp(ArrayRef<Tensor> operand, Region &cond,
+                                Region &body, Scope &scope) {
+  SmallVector<Tensor> runtimeResults(operand);
+
+  auto condResults = eval(cond, operand, &scope);
+  if (condResults.size() != 1)
+    llvm::report_fatal_error("Failed to evaluate cond");
+
+  while (condResults[0].get(*condResults[0].index_begin()).getBooleanValue()) {
+    runtimeResults = eval(body, runtimeResults, &scope);
+    condResults = eval(cond, runtimeResults, &scope);
+    if (condResults.size() != 1)
+      llvm::report_fatal_error("Failed to evaluate cond");
+  }
+
+  return runtimeResults;
+}
+
 Tensor evalXorOp(const Tensor &lhs, const Tensor &rhs, Type resultType) {
   Tensor result(resultType);
   for (auto it = lhs.index_begin(); it != lhs.index_end(); ++it)
     result.set(*it, lhs.get(*it) ^ rhs.get(*it));
   return result;
 }
 
+SmallVector<Tensor> eval(Region &region, ArrayRef<Tensor> args,
+                         Scope *parentScope) {
+  if (!region.hasOneBlock())
+    llvm::report_fatal_error("Expected single block region");
```

Comment:
I'm realizing that we probably don't have to explicitly check this. If there are multiple blocks, then the execution path either doesn't encounter them (good) or the execution path will encounter a cf.br operation or similar and crash (also good).

Reply:
makes sense.